src/routes/preHandlers/payments.js

```diff
@@ -0,0 +1,61 @@
+const Boom = require('boom');
+const Payment = require('../../models/Payment');
+const Customer = require('../../models/Customer');
+const translate = require('../../helpers/translate');
+
+const { language } = translate;
+
+exports.getPayment = async (req) => {
+  try {
+    const payment = await Payment.findOne({ _id: req.params._id }).lean();
+    if (!payment) throw Boom.notFound(translate[language].paymentNotFound);
+
+    return payment;
+  } catch (e) {
+    req.log('error', e);
+    return Boom.isBoom(e) ? e : Boom.badImplementation(e);
+  }
+};
+
+exports.authorizePaymentsUpdate = (req) => {
+  try {
+    const { credentials } = req.auth;
+    const { payment } = req.pre;
+    if (payment.company.toHexString() === credentials.company._id.toHexString()) return null;
+    throw Boom.forbidden();
+  } catch (e) {
+    req.log('error', e);
+    return Boom.isBoom(e) ? e : Boom.badImplementation(e);
+  }
+};
+
+exports.authorizePaymentsCreation = async (req) => {
+  try {
+    const { credentials } = req.auth;
+    for (const payment of req.payload) {
+      const customer = await Customer.findById(payment.customer);
```

Comment:
Lancer une requête pour chaque paiement risque de prendre du temps. Tu peux faire un `map` sur `req.payload` pour n'avoir que les _id des customers puis faire un `Customer.countDocuments` avec un `$in` et la company en query. Tu n'auras plus qu'à comparer la length de ton tableau avec le compteur. S'il y a une différence, tu retournes une 403.